airbyte-cdk/python/airbyte_cdk/sources/declarative/retrievers/simple_retriever.py

```diff
@@ -219,14 +225,13 @@ def read_records(
         stream_slice: Mapping[str, Any] = None,
         stream_state: Mapping[str, Any] = None,
     ) -> Iterable[Mapping[str, Any]]:
-        records_generator = HttpStream.read_records(self, sync_mode, cursor_field, stream_slice, stream_state)
+        # Warning: use self.state instead of the stream_state passed as argument!
```

Comment:
Could you add to these comments an explanation about why?  (probably because it's less confusing since we decide to self-manage state)

Reply:
To make this clearer, you can even put `stream_state` as part of `**kwargs` in all these methods probably and just hide it from method params entirely